app/controllers/data-format.js

```diff
@@ -43,20 +43,42 @@ export default Ember.Controller.extend(sharedActions, {
     };
     return nextFields[this.get('currentField')];
   }),
+  showErrorState: false,
+  errorMessages: [],
+  resetErrorState: function () {
+    Ember.set(this, 'showErrorState', false);
+    Ember.set(this, 'errorMessages', []);
+  },
   actions: {
     goToField: function(field){
       this.set('currentField', field);
     },
     prevField: function() {
+      this.resetErrorState();
       this.set('currentField', this.get('prevField'));
     },
     nextField: function(){
+      this.resetErrorState();
       this.set('currentField', this.get('nextField'));
     },
+    chooseField: function(heading, column){
+      Ember.set(this.model.submission.get('oaFields')[heading], "fields", []);
+      this.model.submission.get('oaFields')[heading].fields.addObject(column);
+      for (var i = 0; i < 2; i++){
+        Ember.set(this.model.submission.get('exampleRows')[i], heading, [this.model.webServiceResponse.source_data.results[i][column]]);
+      }
+      this.resetErrorState();
+    },
     addFunction: function(field, action){
-      Ember.set(this.model.submission.get('oaFields')[field], "function", action);
-      if (action === "join"){
-        this.set('showAdditionalJoinDropdown', true);
+      if (this.model.submission.get('oaFields')[field].fields.length > 0){
+        this.resetErrorState();
+        Ember.set(this.model.submission.get('oaFields')[field], "function", action);
+        if (action === "join"){
+          this.set('showAdditionalJoinDropdown', true);
+        }
+      } else {
```

Comment:
How many error catches are we going to need for this component? If there is more than one, we maybe want to separate the function from whole process to make it clear where error-catching is happening.